packages/gensx-openai/src/index.tsx

```diff
@@ -8,58 +8,77 @@ import {
 } from "openai/resources/index.mjs";
 import { Stream } from "openai/streaming";
 
-// Create a context for OpenAI
-export const OpenAIContext = gsx.createContext<{
-  client?: OpenAI;
-}>({});
+/**
+ * Create a context for an OpenAI client and bind it to a ChatCompletion component. This allows you to have
+ * multiple OpenAI clients that target different OpenAI-compatible API endpoints.
+ *
+ * @param args - The client options
+ * @returns The OpenAI context and the ChatCompletion component
+ */
+export const createOpenAIClientContext = <
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-parameters
+  ModelName extends string = ChatCompletionCreateParams["model"],
+>(
+  args: ClientOptions,
+  namePrefix: string,
+) => {
+  const client = new OpenAI(args);
 
-export const OpenAIProvider = gsx.Component<ClientOptions, never>(
-  "OpenAIProvider",
-  (args) => {
-    const client = new OpenAI(args);
-    return <OpenAIContext.Provider value={{ client }} />;
-  },
-);
+  const OpenAIClientContext = gsx.createContext(client);
 
-// Create a component for chat completions
-export const ChatCompletion = gsx.StreamComponent<ChatCompletionCreateParams>(
-  "ChatCompletion",
-  async (props) => {
-    const context = gsx.useContext(OpenAIContext);
-
-    if (!context.client) {
-      throw new Error(
-        "OpenAI client not found in context. Please wrap your component with OpenAIProvider.",
-      );
+  const BoundChatCompletion = gsx.StreamComponent<
+    ChatCompletionCreateParams & {
+      model: ModelName;
     }
+  >(`${namePrefix}ChatCompletion`, (props) => {
+    const { stream, ...rest } = props;
+    return (
+      <ChatCompletion
+        context={OpenAIClientContext}
+        stream={stream ?? false}
+        {...rest}
+      />
+    );
+  });
 
-    if (props.stream) {
-      const stream = await context.client.chat.completions.create(props);
+  function OpenAIClientProvider(_: gsx.Args<{}, never>) {
+    return <OpenAIClientContext.Provider value={client} />;
+  }
 
-      async function* generateTokens(): AsyncGenerator<
-        string,
-        void,
-        undefined
-      > {
-        for await (const chunk of stream as Stream<ChatCompletionChunk>) {
-          const content = chunk.choices[0]?.delta?.content;
-          if (content) {
-            yield content;
-          }
-        }
-      }
+  return {
+    Provider: OpenAIClientProvider,
+    ChatCompletion: BoundChatCompletion,
```

Comment:
Feels like we should give an output type to this thing, something like `OpenAICompatibleContext`. Then component authors can write reusable components that accept a context: 

```tsx
interface WebResearchProps {
  topic: string;
  researchContext: OpenAICompatibleContext
}
```